src/js/components/service-requests/service-request-detail.js

```diff
@@ -19,92 +24,176 @@ export class ServiceRequestDetail {
+      );
+
+      if (response.street_name || response.street_number || response.suburb) {
         this.$element.append(
           new BasicBlock({
-            title: "Current status",
-            subtitle: SERVICE_REQUEST_STATUS[response.status],
+            title: "Address",
+            subtitle: `${response.street_name} ${response.street_number}, ${response.suburb}`,
           }).render()
         );
+      }
 
+      if (response.coordinates) {
+        const staticMapEvent = new CustomEvent("add-static-map", {
+          detail: response.coordinates,
+        });
         this.$element.append(
-          new BasicBlock({
-            title: "Service area",
-            subtitle: response.type || "No service type specified",
-          }).render()
+          $("<div />", {
+            id: "static-service-request-map",
+            style: "height: 300px",
+          })
         );
-
-        if (response.street_name || response.street_number || response.suburb) {
-          this.$element.append(
-            new BasicBlock({
-              title: "Address",
-              subtitle: `${response.street_name} ${response.street_number}, ${response.suburb}`,
-            }).render()
-          );
-        }
-
-        if (response.coordinates) {
-          const staticMapEvent = new CustomEvent("add-static-map", {
-            detail: response.coordinates,
-          });
-          this.$element.append(
-            $("<div />", {
-              id: "static-service-request-map",
-              style: "height: 300px",
-            })
-          );
-          document.dispatchEvent(staticMapEvent);
-        }
-
-        this.$element.append([
-          $serviceDescriptionHeading,
-          new BlockPreWrap(response.description).render(),
-        ]);
-      })
-      .fail((a, b) => {
-        this.$element.empty().append(
-          new StatusMessage({
-            text: "Error while loading service request details.",
-            status: "failure",
-          }).render()
+        document.dispatchEvent(staticMapEvent);
+      }
+
+      this.$element.append([
+        $serviceDescriptionHeading,
+        new BlockPreWrap(response.description).render(),
+      ]);
+
+      this.$element.append([
+        $serviceRequestImagesHeading,
+        $uploadImagesInput,
+        $uploadImagesClass,
+        $submitButton,
+      ]);
+    });
+
+    function renderFileAlreadyUploaded(blob, fileId) {
+      let url = URL.createObjectURL(blob);
+      const $preview = $uploadImagePreviewTemplate
+        .clone()
+        .attr({
+          id: "upload-image-preview-" + fileId,
+        })
+        .removeClass("hidden");
+
+      $preview.find(".image-preview__remove").remove();
+      $preview.css("background-image", "url('" + url + "')");
+      $uploadImagesClass.append($preview);
+    }
+
+    api.getServiceRequestFiles(serviceRequestId).then((response) => {
+      for (let index in response) {
+        api.getServiceRequestFile(
```

Comment:
I was confused here for a second, not knowing what this all does. When I looked at the actual implementation I noticed that the last parameter is actually a callback function. Perhaps worth adding a comment here? Something like:

```
For each image, the `renderFileAlreadyUploaded` function above is called.
```

Reply:
Added